challenges/functions.js

```diff
@@ -6,25 +6,36 @@
   * The last parameter accepts a callback 
   * In the body of the function return the callback with the two parameters that you created
 */
-
+function consume(arg1, arg2, callback){
+  return callback(arg1, arg2);
+}
 
 /* Step 2: Create several functions to callback with consume();
   * Create a function named add that returns the sum of two numbers
   * Create a function named multiply that returns the product of two numbers 
   * Create a function named greeting that accepts a first and last name and returns "Hello first-name last-name, nice to meet you!"
 */
-
+function add(x, y){
+  return x + y;
+}
+function multiply(x, y){
+  return x * y;
+}
+function greeting(first, last){
+  return `Hello ${first} ${last}, nice to meet you!`
+}
 
 /* Step 3: Check your work by uncommenting the following calls to consume(): */
-//consume(2,2,add); // 4
-// consume(10,16,multiply); // 160
-// consume("Mary","Poppins", greeting); // Hello Mary Poppins, nice to meet you!
+consume(2,2,add); // 4
+consume(10,16,multiply); // 160
+consume("Mary","Poppins", greeting); // Hello Mary Poppins, nice to meet you!
 
 
 // ==== Closures ==== 
 // Explain in your own words why `nestedfunction()` can access the variable `internal`.
 
-// Explanation: 
+Explanation: The nestedFunction is "enclosed" by the function myFunction, and therefore has access to any variables stated in myFunction. It's existance within the scope of it's parent function allow the child to use the resources provided by the parent.
```

Comment:
Would be good to mention `closure` here

Also, this should be commented out, otherwise your code won't run